crates/compiler/mono/src/borrow.rs

```diff
@@ -393,7 +393,11 @@ impl<'state, 'a> State<'state, 'a> {
                 let borrow_signature =
                     match borrow_signatures.procs.get(&(name.name(), proc_layout)) {
                         Some(s) => s,
-                        None => unreachable!("no borrow signature for {name:?} layout"),
+                        None =>  unreachable!(
+                            "\n\tNo borrow signature for {name:?} layout.\n\n\t\
+                            Tip: This can happen when you call a function with less arguments than it expects.\n\t\
+                            Like `Arg.list!` instead of `Arg.list! {{}}`"
```

Comment:
This will be a breaking change to `Arg.list` in the new basic-cli release, so I thought it was worth it to mention it specifically.